### DISCUSSION TITLE
[@types/node] a problem with AbortSignal

### DISCUSSION THREAD

**Discussion by YakovL** [DefinitelyTyped/DefinitelyTyped#64176] [Issues with a @types package] [answered]
To give you some context, I'm not 100% sure this is the issue of `@types/node` as I didn't have this problem until I've introduced Nest.js into my project. Still, on attempt to start Nest, I'm getting (I've added some whitespace for readability):

```
node_modules/@types/node/globals.d.ts:72:13 - error TS2403: Subsequent variable declarations must have the same type.
  Variable 'AbortSignal' must be of type '{ new (): AbortSignal; prototype: AbortSignal; abort(reason?: any): AbortSignal; timeout(milliseconds: number): AbortSignal; }',
  but here has type '{ new (): AbortSignal; prototype: AbortSignal; }'.

72 declare var AbortSignal: {
               ~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2071:13
    2071 declare var AbortSignal: {
                     ~~~~~~~~~~~
    'AbortSignal' was also declared here.
```
which looks pretty much like a problem of either `@types/node` or `typescript` package. I have: "@types/node": "^17.0.45" and "typescript": "^4.9.4" in devDependencies, I've also removed node_modules and package-lock.json and reinstalled the whole thing to be sure it's not some caching/dependency resolving issue.

May be related: #54178, #56716

[a comment by an automated account is not shown]

**Comment by Semigradsky** [comment, marked as answer]
> I have: "@types/node": "^17.0.45"

What about updating node typings? The last version currently is `18.11.18`

**Reply by YakovL** [reply]
Oh, I see, thanks. I've tried `npm update @types/node`, but that only updated the minor version. After `npm i @types/node@18` the error went away.
